src/OrchardCore.Modules/OrchardCore.Workflows/Views/WorkflowTrimming.Fields.Edit.cshtml

```diff
@@ -27,87 +27,44 @@
-            <label class="form-check-label" for="@Html.IdFor(m => m.Statuses[3])">
-                @T["Executing"]
-            </label>
-        </div>
-    </div>
-
-    <div class="col-xs-12">
-        <div class="form-check">
-            <input type="hidden" asp-for="@Model.Statuses[4].Status" />
-            <input class="form-check-input workflow-status" type="checkbox" asp-for="@Model.Statuses[4].IsSelected" id="@Html.IdFor(m => m.Statuses[4])">
-            <label class="form-check-label" for="@Html.IdFor(m => m.Statuses[4])">
-                @T["Halted"]
-            </label>
-        </div>
-    </div>
+@{
+    var statusIterator = 0;
+    var localizedStatusNames = new Dictionary<WorkflowStatus, LocalizedHtmlString>
+    {
+        { WorkflowStatus.Idle, T["Idle"] },
+        { WorkflowStatus.Starting, T["Starting"] },
+        { WorkflowStatus.Resuming, T["Resuming"] },
+        { WorkflowStatus.Executing, T["Executing"] },
+        { WorkflowStatus.Halted, T["Halted"] },
+        { WorkflowStatus.Finished, T["Finished"] },
+        { WorkflowStatus.Faulted, T["Faulted"] },
+        { WorkflowStatus.Aborted, T["Aborted"] },
+    };
+}
 
-    <div class="col-xs-12">
-        <div class="form-check">
-            <input type="hidden" asp-for="@Model.Statuses[5].Status" />
-            <input class="form-check-input workflow-status" type="checkbox" asp-for="@Model.Statuses[5].IsSelected" id="@Html.IdFor(m => m.Statuses[5])">
-            <label class="form-check-label" for="@Html.IdFor(m => m.Statuses[5])">
-                @T["Finished"]
-            </label>
-        </div>
-    </div>
-
-    <div class="col-xs-12">
-        <div class="form-check">
-            <input type="hidden" asp-for="@Model.Statuses[6].Status" />
-            <input class="form-check-input workflow-status" type="checkbox" asp-for="@Model.Statuses[6].IsSelected" id="@Html.IdFor(m => m.Statuses[6])">
-            <label class="form-check-label" for="@Html.IdFor(m => m.Statuses[6])">
-                @T["Faulted"]
-            </label>
+<div class="mb-3 row">
+    <label asp-for="Statuses" class="control-label">@T["Statuses"]</label>
+    @foreach (var status in Enum.GetValues<WorkflowStatus>())
+    {
+        <div class="col-xs-12">
+            <div class="form-check">
+                <input type="hidden" asp-for="@Model.Statuses[statusIterator].Status" />
+                <input class="form-check-input workflow-status" type="checkbox" id="@Html.IdFor(m => m.Statuses[statusIterator])"
+                       asp-for="@Model.Statuses[statusIterator].IsSelected">
+                <label class="form-check-label" for="@Html.IdFor(m => m.Statuses[statusIterator])">
+                    @if (localizedStatusNames.TryGetValue(status, out var localizedStatus))
+                    {
+                        @localizedStatus
```

Comment:
Since it's only used there, we usually do a switch/case on the enum and put the localization in-place